src/utils/utils.h

```diff
@@ -9,6 +9,8 @@
 #include "reduce.h"
 #include <stdlib.h>
 
+#define MIN(_a, _b) ((_a < _b) ? _a : _b)
```

Comment:
you may get the notorious "double evaluation" bug there. plz define the macro as suggested there : https://stackoverflow.com/questions/3437404/min-and-max-in-c